### PR TITLE
add support for tmux

### PR DESCRIPTION
**Pull request by getchoo**
this adds support for the [tmux](https://github.com/catppuccin/tmux) theme

### CODE REVIEW

File: modules/home-manager/tmux.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.programs.tmux.catppuccin;
+
+  plugin = with builtins;
+    with pkgs;
+    let rev = "4e48b09a76829edc7b55fbb15467cf0411f07931";
+    in pkgs.tmuxPlugins.mkTmuxPlugin {
```
Comment (Stonks3141) [inline modules/home-manager/tmux.nix:8]:
```suggestion
    in tmuxPlugins.mkTmuxPlugin {
```
`pkgs` is already in scope.

File: modules/home-manager/tmux.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      pluginName = "catppuccin";
+      version = substring 0 7 rev;
```
Comment (Stonks3141) [inline modules/home-manager/tmux.nix:10]: I assume this is a tmux thing, but can we use the full rev?

File: modules/home-manager/tmux.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      default = config.catppuccin.flavour;
+      description = "Catppuccin flavour for tmux";
+    };
+  };
+
+  config.programs.tmux.plugins = with lib;
+    mkIf cfg.enable [{
+      inherit plugin;
+      extraConfig = "set -g @catppuccin_flavour '${cfg.flavour}'";
+    }];
```
Comment (Stonks3141) [inline modules/home-manager/tmux.nix:32]:
```suggestion
  config.programs.tmux.plugins = with lib; mkIf cfg.enable [
    {
      inherit plugin;
      extraConfig = "set -g @catppuccin_flavour '${cfg.flavour}'";
    }
  ];
```
I think the new line for `cfg.enable` and `[{` just looks a bit gross. Ignore this if this code is after a formatter ran on it.

**Review by Stonks3141 — CHANGES_REQUESTED** [review]

File: modules/home-manager/tmux.nix (hunk 2 above)
Comment (getchoo) [reply modules/home-manager/tmux.nix:10]: it isn't. most packages just slim down the commit hash or use another form of shorthand like a date, plus it's also what i usually do

Comment (Stonks3141) [reply modules/home-manager/tmux.nix:10]: Oh, I didn't realize that was the version of the package, not the version to fetch. :)

**Review by Stonks3141 — APPROVED** [review]

**Pull request merged**